arches_search/utils/advanced_search/related_clause_evaluator.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, Optional
+from django.db.models import Exists, OuterRef, Q
+
+QUANTIFIER_ANY = "ANY"
+QUANTIFIER_ALL = "ALL"
+QUANTIFIER_NONE = "NONE"
+
+CLAUSE_TYPE_RELATED = "RELATED"
+
+
+class RelatedClauseEvaluator:
+    def __init__(
+        self, search_model_registry, facet_registry, path_navigator, predicate_builder
+    ) -> None:
+        self.search_model_registry = search_model_registry
+        self.facet_registry = facet_registry
+        self.path_navigator = path_navigator
+        self.predicate_builder = predicate_builder
+
+    def evaluate(
+        self,
+        mode: str,
+        clause_payload: Dict[str, Any],
+        traversal_context: Optional[Dict[str, Any]] = None,
+        terminal_datatype_name: Optional[str] = None,
+    ):
+        if mode == "anchor":
+            return self._build_anchor_presence_exists(
+                clause_payload, terminal_datatype_name
+            )
+        if mode == "child":
+            if traversal_context is None:
+                raise ValueError("traversal_context is required for mode='child'")
```

Comment:
You might have reasons for not using a translation, but I thought I'd add the suggestion in case it was just an oversight
```suggestion
                raise ValueError("traversal_context is required for mode='child'")
                raise ValueError(_("traversal_context is required for mode='child'"))
```